web-services/query/src/main/java/datawave/webservice/query/runner/RunningQuery.java

```diff
@@ -203,22 +204,22 @@ public ResultsPage next() throws Exception {
         List<Object> resultList = new ArrayList<>();
         boolean hitPageByteTrigger = false;
         boolean hitPageTimeTrigger = false;
+        boolean hitIntermediateResult = false;
         try {
             addNDC();
             int currentPageCount = 0;
             long currentPageBytes = 0;
             
             // test for any exceptions prior to loop as hasNext() would likely be false;
             testForUncaughtException(resultList.size());
-            
             while (!this.finished && ((future != null) || this.iter.hasNext())) {
```

Comment:
I am missing something here.  As this is where the beginning of a page actually starts, should there not be a call to set the page start time here?

Reply:
The timeout is reset in the GroupingTransform at the same time that the intermediate result flag is set.(client side) since that's what actually knows both if long running queries are allowed and if the time has elapsed.  (all the transformers depend on knowing this information -- in the case of grouping, this has to be tested before dciding to pop a document from the key list)  Since the RunningQuery detects that `hitIntermediateResult` flag, the RunningQuery  knows that a new page should be started (when next() is called again)

The running query doesn't know when the first query start time was set (which gets set on the transformers in markQueryStarted())  I understand that kicking off the query is not the exact same as when the results for a page will start to be returned, but to add access to the transformers would requiring adding a dependency to datawave-query-core, AND keeping that variable in sync between the transformers (which will still and to know it) and the RuningQuery.